### PR TITLE
feat: say bye to TravisCI, say hi to CircleCI

### PR DESCRIPTION
**Pull request by dkontorovskyy**
- [ ] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
After migrating AppVeyor Windows tests to CirlceCI, it's finally time for the last step and migrate everything to CircleCI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,11 +26,11 @@
     "find-circular": "npm run build && madge --circular ./dist",
     "format": "prettier --write '{src,test,scripts}/**/*.{js,ts}'",
     "prepare": "npm run build",
-    "test:common": "npm run check-tests && npm run build && npm run lint && node --require ts-node/register src/cli test --org=snyk",
```
Comment (dkontorovskyy) [inline package.json:29]: Build runs in Circle step before. No need to re-run

File: package.json [hunk 2]
```diff
@@ -26,11 +26,11 @@
...
+    "test:common": "npm run check-tests && npm run lint && node --require ts-node/register src/cli test --org=snyk",
     "test:acceptance": "tap test/acceptance/**/*.test.* -Rspec --timeout=300 --node-arg=-r --node-arg=ts-node/register",
     "test:system": "tap test/system/*.test.* -Rspec --timeout=300 --node-arg=-r --node-arg=ts-node/register",
     "test:test": "tap test/*.test.* -Rspec --timeout=300 --node-arg=-r --node-arg=ts-node/register",
-    "test": "npm run test:common && npm run test:acceptance && npm run test:system && npm run test:test",
```
Comment (dkontorovskyy) [inline package.json:33]: No need to lint in every test run. Do it ones and before all tests

File: .circleci/config.yml [hunk 3]
```diff
@@ -64,11 +71,40 @@ commands:
           command: npm --version
 
 jobs:
-  test-all:
+  common:
+    <<: *defaults
+    docker:
+      - image: circleci/node:<< parameters.node_version >>
```
Comment (gitphill) [inline .circleci/config.yml:77]: fancy, and this is configured in circle?

Comment (dkontorovskyy) [reply .circleci/config.yml:77]: It's configured couple lines down :) 

https://github.com/snyk/snyk/pull/1072/files#diff-1d37e48f9ceff6d8030570cd36286a61R166

File: test/proxy.test.js [hunk 4]
```diff
@@ -54,11 +58,18 @@ test('request respects proxy environment variables', function(t) {
       .catch((err) => t.fail(err.message))
       .then(() => {
         proxy.close();
+
+        // Restore env variables
+        process.env.NO_PROXY = tmpNoProxy;
```
Comment (gitphill) [inline test/proxy.test.js:63]: t.teardown is probably a more reliable way for restoring

File: test/proxy.test.js [hunk 5]
```diff
@@ -70,6 +81,9 @@ test('request respects proxy environment variables', function(t) {
...
```
Comment (gitphill) [inline test/proxy.test.js:86]: t.teardown is probably a more reliable way for restoring

File: test/request.test.ts [hunk 6]
```diff
@@ -335,6 +340,9 @@ test('request with http proxy calls needle as expected', (t) => {
         ),
         'needle called as expected',
       );
+
+      // Restore env variables
+      process.env.NO_PROXY = tmpNoProxy;
```
Comment (gitphill) [inline test/request.test.ts:345]: t.teardown is probably a more reliable way for restoring

**Pull request merged**